test/r_tree/r_tree_test.dart

```diff
@@ -269,60 +269,127 @@ main() {
   });
 }
 
-void assertTreeHeightValidity<E>(RTree<E> tree) {
+/// Comprehensively assert the consistency of the specified tree, including node height, parent references, and bounding
+/// rectangles.
+void assertTreeValidity<E>(RTree<E> tree) {
   try {
-    assertNodeHeightValidity(tree, tree.currentRootNode);
+    assertNodeValidity(tree, tree.currentRootNode);
   } on StateError catch (e) {
     fail('${e.message}\nTree:\n${stringifyTree(tree)}');
   }
 }
 
-int assertNodeHeightValidity<E>(RTree<E> tree, RTreeContributor contributor) {
+/// Comprehensively assert the consistency of the specified subtree, including node height, parent references, and
+/// bounding rectangles.
+_SubtreeValidationData assertNodeValidity<E>(RTree<E> tree, RTreeContributor contributor) {
   if (contributor is LeafNode<E>) {
-    if (contributor.height != 1) {
-      throw StateError('Leaf height of ${contributor.height} should be 1.');
-    }
-
-    return 1;
+    return assertLeafNodeValidity(tree, contributor);
   } else if (contributor is NonLeafNode<E>) {
-    var maxChildHeight = 0;
-    if (contributor.children.isNotEmpty) {
-      for (final child in contributor.children) {
-        final childHeight = assertNodeHeightValidity(tree, child);
-        if (childHeight > maxChildHeight) {
-          maxChildHeight = childHeight;
-        }
+    return assertNonLeafNodeValidity(tree, contributor);
+  }
+
+  // This is a datum
+  return _SubtreeValidationData(0, contributor.rect);
+}
+
+/// Comprehensively assert the consistency of the subtree rooted at the specified leaf node, including node height,
+/// parent references, and bounding rectangles.
+_SubtreeValidationData assertLeafNodeValidity<E>(RTree<E> tree, LeafNode<E> node) {
+  if (node.height != 1) {
+    throw StateError('Leaf height of ${node.height} should be 1.');
+  }
+
+  const defaultRect = Rectangle<num>(0, 0, 0, 0);
+  var actualRect = defaultRect;
+
+  if (node.children.isNotEmpty) {
+    for (final child in node.children) {
```

Comment:
this is subjective but I've become a fan of doing this sort of thing using first/skip:
```dart
actualRect = node.children.first
for (final child in node.children.skip(1)) {
```

Reply:
I wonder if we should be using a util method that shared test and prod code instead of re-implementing here